app.js

```diff
@@ -0,0 +1,73 @@
+const express = require('express');
+const path = require('path');
+const mongoose = require('mongoose');
+const ExpressError = require('./utils/ExpressError');
+const methodOverride = require('method-override');
+const ejsMate = require('ejs-mate');
+const session = require('express-session');
+const flash = require('connect-flash');
+
+const userRoutes = require('./routes/userRoutes');
+const bookRoutes = require('./routes/bookRoutes');
+const reviewRoutes = require('./routes/reviewRoutes');
+
+mongoose.connect('mongodb://127.0.0.1:27017/friends-shelves');
+
+const db = mongoose.connection;
+db.on("error", console.error.bind(console, "connection error:"));
+db.once("open", () => {
+    console.log('Database connected');
+});
+
+const app = express();
+
+app.engine('ejs', ejsMate);
+app.set('view engine', 'ejs');
+app.set('views', path.join(__dirname, 'views'));
+
+app.use(express.urlencoded({ extended: true }));
+app.use(methodOverride('_method'));
+app.use(express.static(path.join(__dirname, 'public')));
+
+const sessionConfig = {
```

Comment:
This is a good start. The next iteration could be using an environment configuration aka the famous `.env` file to make sure that you don't put your secrets in version control where they can be discovered by others.

A good library to do that is https://github.com/motdotla/dotenv